### DISCUSSION TITLE
How to convert the timestamp in the birdwatcher checkpoint api to the current unix timestamp

### DISCUSSION THREAD

**Discussion by TonyAnn** [milvus-io/milvus#35812] [Q&A and General discussion] [answered]
The timestamp in the birdwatcher checkpoint api is 452136031162728450, how to compare it with the current time
![birdwatcher checkpoint timestame](https://github.com/user-attachments/assets/b19ddae2-0c55-4e2c-9099-78de0b919bce)

**Comment by yanliang567** [comment]
@congqixia please help to take a look

**Comment by congqixia** [comment, marked as answer]
This timestamp is milvus logic ts value, you could convert it into time with following logic:
```
const (
	logicalBits     = 18
	logicalBitsMask = (1 << logicalBits) - 1
)

func ParseTS(ts uint64) (time.Time, uint64) {
	logical := ts & logicalBitsMask
	physical := ts >> logicalBits
	physicalTime := time.Unix(int64(physical/1000), int64(physical)%1000*time.Millisecond.Nanoseconds())
	return physicalTime, logical
}
```

**Reply by TonyAnn** [reply]
Awesome, thanks for your support
